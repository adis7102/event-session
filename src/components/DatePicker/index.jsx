import React from 'react';
import cn from 'classnames';

const DatePicker = (props) => {
  const { name, title, onChange, onReset, value } = props;

  return (
    <div className="input">
      <p className="title">{title}</p>
      <div className="input-wrap">
        <input
          type="date"
          name={name}
          onChange={(e) => onChange(e?.target?.value)}
          value={value}
          data-testid="date-picker"
        />
        <div
          className={cn("reset-button", {
            disabled: !value
          })}
          onClick={() => onReset()}
        >
          <p>reset date</p>
        </div>
      </div>
    </div>
  )
}

export default DatePicker