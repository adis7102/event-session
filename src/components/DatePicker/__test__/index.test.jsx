import React from "react";
import DatePicker from '../index';

import { render } from "@testing-library/react";

it('should render Input correctly', () => {
    const { container } = render(<DatePicker />);
  
    expect(container).toMatchSnapshot();
});