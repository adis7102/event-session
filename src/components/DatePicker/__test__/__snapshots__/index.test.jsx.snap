// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input correctly 1`] = `
<div>
  <div
    class="input"
  >
    <p
      class="title"
    />
    <div
      class="input-wrap"
    >
      <input
        data-testid="date-picker"
        type="date"
        value=""
      />
      <div
        class="reset-button disabled"
      >
        <p>
          reset date
        </p>
      </div>
    </div>
  </div>
</div>
`;
